src/libraries/System.Text.Json/src/System/Text/Json/JsonHelpers.Date.cs

```diff
@@ -211,12 +211,6 @@ private static bool TryParseDateTimeOffset(ReadOnlySpan<byte> source, out DateTi
         {
             parseData = default;
 
-            // Source does not have enough characters for YYYY-MM-DD
```

Comment:
Please add a `Debug.Assert` with this check and a comment saying we've validated this upstream.

Reply:
thanks for your review

it has already been tested from https://github.com/dotnet/runtime/blob/ff94c801083aae2bb9aa9fd8f57443a9606c585e/src/libraries/System.Text.Json/tests/Utf8JsonReaderTests.TryGet.Date.cs#L127 

with ```InvalidISO8601Tests``` cases you said.

so i've jsut added ``` Debug.Assert(source.Length >= 10);```  2384fa91363b46258d724f122db2eac387f0f9f4

therefore invalid test cases that length is under 10 will be asserted.

Reply:
My point in https://github.com/dotnet/runtime/pull/40759#pullrequestreview-466801520 was that the first few lines of [`InvalidISO8601Tests`](https://github.com/dotnet/runtime/blob/7a2d5d28e6cfd615a59021faf05672b937896dba/src/libraries/System.Text.Json/tests/JsonDateTimeTestData.cs#L106) should look like this 

```cs
 public static IEnumerable<object[]> InvalidISO8601Tests()
{
  // Too short
  yield return new object[] { "\"997-07-16\"" };
  yield return new object[] { "\"1997-07\"" };
  ...

  // Invalid YYYY-MM-DD
  yield return new object[] { "\"0997 07-16\"" };
  yield return new object[] { "\"0997-0a-16\"" };
  yield return new object[] { "\"0997-07 16\"" };
  yield return new object[] { "\"0997-07-160997-07-16\"" };
  yield return new object[] { "\"0997-07-16abc\"" };
  yield return new object[] { "\"0997-07-16 \"" };
  yield return new object[] { "\"0997-07-16,0997-07-16\"" };
  yield return new object[] { "\"1997-07-16T19:20abc\"" };
  yield return new object[] { "\"1997-07-16T19:20, 123\"" };

  ...
}
```

where some test cases were moved from the `// Invalid YYYY-MM-DD` section.

Reply:
i read it wrong 😅
Sorry to bother you

I've moved some test cases that are too short under ```//Too short``` .
you can check it here https://github.com/dotnet/runtime/commit/be80a667ccbb404a6be329ebd2404e033cddc0a8